### PR TITLE
Current location

### LINKED ISSUE #386: Grab Current Location
image
image

- Geolocation API
- Only show the city where the user is in, do not show an address, should just be a City and a State/Province

### PR DESCRIPTION
**Pull request by Zorten**
Grabs current city and region and displays it as text after a tap on the map pin icon on the create screen. Closes #386.

photo_2024-08-14_01-51-41
photo_2024-08-14_01-51-39

### CODE REVIEW

File: components/create/details.tsx [hunk 1]
```diff
@@ -40,12 +47,35 @@ const Details = ({
   addTag,
 }: props) => {
   const [tag, setTag] = useState("");
+  const [location, setLocation] = useState<string | null>();
+  const [errorMsg, setErrorMsg] = useState<string | null>();
+  const [isLoading, setIsLoading] = useState(false);
```
Comment (shahdivyank) [inline components/create/details.tsx:52]: call it just loading

File: components/create/details.tsx [hunk 2]
```diff
@@ -40,12 +47,35 @@ const Details = ({
   addTag,
 }: props) => {
   const [tag, setTag] = useState("");
+  const [location, setLocation] = useState<string | null>();
+  const [errorMsg, setErrorMsg] = useState<string | null>();
```
Comment (shahdivyank) [inline components/create/details.tsx:51]: call it just error

Comment (shahdivyank) [inline components/create/details.tsx:51]: types should just be string, u can set empty string as the default

File: components/create/details.tsx [hunk 3]
```diff
@@ -40,12 +47,35 @@ const Details = ({
...
+  const [isLoading, setIsLoading] = useState(false);
 
   const handleAdd = (value: string) => {
     addTag(value);
     setTag("");
   };
 
+  const getCurrLocation = async () => {
+    setIsLoading(true);
+    let { status } = await Location.requestForegroundPermissionsAsync();
```
Comment (shahdivyank) [inline components/create/details.tsx:61]: const

File: components/create/details.tsx [hunk 4]
```diff
@@ -40,12 +47,35 @@ const Details = ({
...
+    if (status !== "granted") {
+      setErrorMsg("Permission Denied");
+      return;
+    }
+
+    let location = await Location.getCurrentPositionAsync({});
```
Comment (shahdivyank) [inline components/create/details.tsx:67]: const

File: components/create/details.tsx [hunk 5]
```diff
@@ -40,12 +47,35 @@ const Details = ({
...
+
+    //get city and state
+    let reverseGeocode = await Location.reverseGeocodeAsync({
```
Comment (shahdivyank) [inline components/create/details.tsx:70]: cons

**Comment by Zorten** [comment]
image

**Review by shahdivyank — APPROVED** [review]

**Pull request merged**